File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/data/DataFloat64.java
package tasks.data;

import com.systemincloud.modeler.tasks.javatask.api.InputPort;
import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.InputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Float64;

@JavaTaskInfo
public class DataFloat64 extends JavaTask {

	@InputPortInfo(name = "In", dataType = Float64.class)
	public InputPort in;
	@OutputPortInfo(name = "Out", dataType = Float64.class)
	public OutputPort out;

	@Override
	public void execute() {
		Float64 inFloat64 = in.getData(Float64.class);
		out.putData(new Float64(inFloat64.getValue()/2));
	}

}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/data/DataFloat64.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.InputPort import InputPort
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Float64 import Float64

@PythonTaskInfo
class DataFloat64(PythonTask):

    def __init_ports__(self):
        self.in1 = InputPort(name="in1", data_type=Float64)
        self.in2 = InputPort(name="in2", data_type=Float64)
        self.out = OutputPort(name="out", data_type=Float64)

    def execute(self, grp):
        in1Data = self.in1.get_data(Float64)
        in2Data = self.in2.get_data(Float64)
        self.out.put_data(Float64(in1Data.value + in2Data.value))
